web-registry/src/components/organisms/ChooseCreditClass/ClassOptions.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { ChooseClassTemplate, CreditClassCard } from './ChooseClassTemplate';
+import { useGetCreditClassOptions } from './hooks/useGetCreditClassOptions';
+import { validate, getProjectPageBaseData } from '../../../lib/rdf';
+import {
+  useUpdateProjectByIdMutation,
+  useShaclGraphByUriQuery,
+} from '../../../generated/graphql';
+import { CreditClassMetadataLD } from '../../../generated/json-ld';
+
+const ClassOptions: React.FC = () => {
+  const [error, setError] = useState('');
+  const navigate = useNavigate();
+  const { projectId } = useParams();
+  const creditClassOptions = useGetCreditClassOptions();
+  const { data: graphData } = useShaclGraphByUriQuery({
+    variables: {
+      uri: 'http://regen.network/ProjectPageShape',
+    },
+  });
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  // Reset error message, so it reappears when user tries again
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => {
+        setError('');
+      }, 5000);
+    }
+  }, [error]);
+
+  async function handleSelection(
+    creditClassId: string,
+    creditClassOnChainId: string,
+    creditClassMetadata: Partial<CreditClassMetadataLD>,
+  ): Promise<void> {
+    if (graphData?.shaclGraphByUri?.graph) {
+      const metadata = getProjectPageBaseData(creditClassOnChainId);
+
+      metadata['regen:vcsMethodology'] = {
+        'schema:name': creditClassMetadata['schema:name'],
+        'schema:url': creditClassMetadata?.['schema:url'],
+      };
+
```

Comment:
I believe the above is all we need for VCS metadata, but LMK if there should be more. I referenced this:
https://github.com/regen-network/regen-registry-standards/blob/main/jsonld/projects/C01-verified-carbon-standard-project.json

Reply:
removed